staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go

```diff
@@ -103,7 +103,17 @@ func UpdateResource(r rest.Updater, scope *RequestScope, admit admission.Interfa
 		original := r.New()
 
 		trace.Step("About to convert to expected version")
-		decoder := scope.Serializer.DecoderToVersion(s.Serializer, scope.HubGroupVersion)
+		decodeSerializer := s.Serializer
```

Comment:
is this use-case common enough to warrant some kind of internal helper like `decodeSerializer := serializerFor(req)`, which also takes care of the feature gate?

Reply:
I think it’s a little more involved given the serializer requirements of the other handlers (specifically the various ways patch handling works). I’m going to follow up on this in another commit that tries to better unify decode serializer generation across all the handlers.